src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -627,11 +792,29 @@ public ProcessExecutionResults RunApkInstrumentation(string apkName, string? ins
         return result;
     }
 
+    private static AndroidDevice DeviceFromPropertyBag(KeyValuePair<string, Dictionary<string, string?>> properties)
+        => new(DeviceSerial: properties.Key)
+        {
+            Architecture = properties.Value.GetValueOrDefault(AdbPropertyNames.Architecture, null),
+
+            ApiVersion = properties.Value.TryGetValue(AdbPropertyNames.ApiVersion, out string? apiVersion) && apiVersion is not null
+                ? int.Parse(apiVersion)
+                : null,
+
+            InstalledApplications = properties.Value.TryGetValue(AdbPropertyNames.InstalledApps, out string? apps) && apps is not null
+                ? apps.Split("\n")
+                : null,
+
+            SupportedArchitectures = properties.Value.TryGetValue(AdbPropertyNames.SupportedArchitectures, out string? archs) && archs is not null
+                ? archs.Split(new char[] { ',', '\r', '\n' })
+                : null,
+        };
+
     #endregion
 
-    #region Process runner helpers
+        #region Process runner helpers

```

Comment:
Unnecessary indentation
```suggestion
    #region Process runner helpers

```